kfp/kfp_support_lib/shared_workflow_support/src/runtime_utils/remote_jobs_utils.py

```diff
@@ -121,41 +121,37 @@ def create_ray_cluster(
         """
         # start with templates
         # head_node
-        cpus = head_node.get("cpu", 1)
-        memory = head_node.get("memory", 1)
-        gpus = head_node.get("gpu", 0)
-        accelerator = head_node.get("gpu_accelerator", None)
+        dct = {}
+        dct["cpu"] = head_node.get("cpu", 1)
+        dct["memory"] = head_node.get("memory", 1)
+        dct["gpu"] = head_node.get("gpu", 0)
+        dct["gpu_accelerator"] = head_node.get("gpu_accelerator", None)
         head_node_template_name = f"{name}-head-template"
-        _, _ = self.api_server_client.delete_compute_template(ns=namespace, name=head_node_template_name)
-        head_template = Template(
-            name=head_node_template_name,
-            namespace=namespace,
-            cpu=cpus,
-            memory=memory,
-            gpu=gpus,
-            gpu_accelerator=accelerator,
-        )
+        dct["name"]=head_node_template_name
+        dct["namespace"]=namespace
+        if "tolerations" in head_node:
+            dct["tolerations"] = head_node.get("tolerations")
```

Comment:
tolerations is a list of dictionaries. How do you pass it as parameter to workflow?

Reply:
It was added to ray_head_options/ray_worker_options in the workflow as follows:
```
{
  "cpu": 1,
  "image": "quay.io/dataprep1/data-prep-kit/noop-ray:latest",
  "memory": 4,
  "tolerations": {
    "key": "preprocessing",
    "operator": "Equal",
    "value": "dedicated",
    "effect": "NoSchedule"
  }
}
```

Reply:
ok